drivers/SmartThings/zigbee-thermostat/src/aqara/init.lua

```diff
@@ -0,0 +1,277 @@
+  if preferences ~= nil then
+    for id, attr in pairs(preference_map) do
+      local old_value = old_preferences[id]
+      local value = preferences[id]
+      if value ~= nil and value ~= old_value then
+        if attr.value_map ~= nil then
+          value = attr.value_map[value]
+        end
+        if preferences[id] == device.preferences["stse.antifreezeModeSetting"] then
+          device:send(cluster_base.write_manufacturer_specific_attribute(device, attr.cluster_id, attr.attribute_id,
+          attr.mfg_code, attr.data_type, value_conversion(value)))
+        end
+        if preferences[id] == device.preferences["stse.notificationOfValveTest"] then
+          device:send(cluster_base.write_manufacturer_specific_attribute(device, attr.cluster_id, attr.attribute_id,
+          attr.mfg_code, attr.data_type, value))
+        end
+      end
+    end
+  end
+end
+
+local function is_aqara_products(opts, driver, device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function supported_thermostat_modes_handler(driver, device, value)
+  device:emit_event(capabilities.thermostatMode.supportedThermostatModes({
+    capabilities.thermostatMode.thermostatMode.manual.NAME,
+    capabilities.thermostatMode.thermostatMode.antifreezing.NAME
+  }, { visibility = { displayed = false } }))
+end
+
+local function do_refresh(driver, device)
+  device:send(Thermostat.attributes.OccupiedHeatingSetpoint:read(device))
+  device:send(cluster_base.read_manufacturer_specific_attribute(device,
+  PRIVATE_CLUSTER_ID, PRIVATE_THERMOSTAT_OPERATING_MODE_ATTRIBUTE_ID, MFG_CODE))
+  device:send(cluster_base.read_manufacturer_specific_attribute(device,
+  PRIVATE_CLUSTER_ID, PRIVATE_VALVE_RESULT_CALIBRATION_ID, MFG_CODE))
+end
+
+local function device_init(driver, device)
+  do_refresh(driver, device)
+end
+
+local function device_added(driver, device)
+  supported_thermostat_modes_handler(driver, device, nil)
+  device:emit_event(capabilities.thermostatHeatingSetpoint.heatingSetpoint({value = 21.0, unit = "C"}))
+  device:emit_event(capabilities.temperatureMeasurement.temperature({value = 27.0, unit = "C"}))
+  device:emit_event(capabilities.thermostatMode.thermostatMode.manual())
+  device:emit_event(capabilities.valve.valve.open())
+  device:emit_component_event(device.profile.components.ChildLock, capabilities.lock.lock.unlocked())
+  device:emit_event(capabilities.hardwareFault.hardwareFault.clear())
+  device:emit_event(valveCalibration.calibrationState.calibrationPending())
+  device:emit_event(invisibleCapabilities.invisibleCapabilities({""}))
+  device:emit_event(capabilities.battery.battery(100))
+end
+
+local function thermostat_alarm_status_handler(driver, device, value, zb_rx)
+  if value.value == 1 then
+    capabilities.hardwareFault.hardwareFault.detected()
```

Comment:
```suggestion
    device:emit_event(capabilities.hardwareFault.hardwareFault.detected())
```

This was the actual issue. You don't need to include the `hardwareFault` yaml, it's actually already in the release artifact.

Reply:
Thanks,modified